targets/TARGET_Samsung/security_subsystem/drivers/sss_driver_sha2.c

```diff
@@ -0,0 +1,199 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/*************** Include Files ********************************************/
+#include "sss_driver_error.h"
+#include "sss_driver_sha2.h"
+
+/*************** Assertions ***********************************************/
+
+/*************** Definitions / Macros *************************************/
+
+/*************** New Data Types (Basic Data Types)  ***********************/
+
+/*************** New Data Types *******************************************/
+
+/*************** Constants ************************************************/
+
+/*************** Variable declarations ************************************/
+
+/*************** Functions  ***********************************************/
+
+
+int sss_SHA2_256(
+    const stOCTET_STRING *pstMessage,
+    stOCTET_STRING *pstDigest)
+{
+
+    int ret = FAIL;
+
+    stOCTET_STRING stHASH_Input;
+    unsigned int object_id;
+    unsigned int block_byte_len;
+
+    //! step 0 : clear Mailbox
+    ret = mb_system_clear(CLEAR_TYPE_MAILBOX);
+
+    if (ret != SSSR_SUCCESS) {
+        return ret;
+    }
+
+    //! step 0 : Check validity of parameter "object_id"
+#if 0
+    if (pstMessage->u32DataByteLen == 0) {
+        return ERROR_HASH_INVALID_LEN_MSG;
+    }
+#endif
```

Comment:
Should this code be commented out or is the check needed? If it isn't needed, can we delete it?

Reply:
It's removed.